tools/c7n_gcp/c7n_gcp/actions/notify.py

```diff
@@ -0,0 +1,121 @@
+              - resource-creator
+              - email@address
+             owner_absent_contact:
+              - other_email@address
+             # which template for the email should we use
+             template: policy-template
+             transport:
+               type: pubsub
+               topic: your-notify-topic
+    """
+    batch_size = 1000
+
+    schema = {
+        'type': 'object',
+        'anyOf': [
+            {'required': ['type', 'transport', 'to']},
+            {'required': ['type', 'transport', 'to_from']}],
+        'properties': {
+            'type': {'enum': ['notify']},
+            'to': {'type': 'array', 'items': {'type': 'string'}},
+            'owner_absent_contact': {'type': 'array', 'items': {'type': 'string'}},
+            'to_from': ValuesFrom.schema,
+            'cc': {'type': 'array', 'items': {'type': 'string'}},
+            'cc_from': ValuesFrom.schema,
+            'cc_manager': {'type': 'boolean'},
+            'from': {'type': 'string'},
+            'subject': {'type': 'string'},
+            'template': {'type': 'string'},
+            'transport': {
+                'oneOf': [
+                    {'type': 'object',
+                     'required': ['type', 'topic'],
+                     'properties': {
+                         'topic': {'type': 'string'},
+                         'type': {'enum': ['pubsub']},
+                     }}],
+            },
+        }
+    }
+
+    @staticmethod
+    def register_notify_action(registry, _):
+        for resource in registry.keys():
+            klass = registry.get(resource)
+            klass.action_registry.register('notify', Notify)
+
+    def process(self, resources, event=None):
+        self.session = utils.local_session(self.manager.session_factory)
+        self.client = self.session.client('pubsub', 'v1', 'projects.topics')
+        project = self.session.get_default_project()
+        message = {
+            'event': event,
+            'account_id': project,
+            'account': project,
+            'region': 'all',
+            'policy': self.manager.data
+        }
+
+        message['action'] = self.expand_variables(message)
+
+        for batch in utils.chunks(resources, self.batch_size):
+            message['resources'] = batch
+            self.publish_message(message)
+
+    # Methods to handle GCP Pub Sub topic publishing
+    def publish_message(self, message):
+        """Publish message to a GCP pub/sub topic
+         """
+        topic = self.get_topic_param()
+
+        try:
+            return self.client.execute_command('publish', {
+                'topic': topic,
+                'body': {
+                    'messages': {
+                        'data': self.pack(message)
+                    }
+                }
+            })
+        except HttpError as e:
+            self.log.error(e)
+
+    def get_topic_param(self, topic=None, project=None):
+        return 'projects/{}/topics/{}'.format(
```

Comment:
this should be removed, and the doc string updated to give an example topic qualified name via relative resource, and the code in publish should use it as is. it will be common to create a single topic across projects, so attempting to format here actually removes functionality for a common case.

Reply:
Good catch. I was thinking single project due to local dev instead of multi-project environments. Fixed in upcoming commit.